keanu-project/src/test/java/io/improbable/keanu/tensor/ScalarTensorShapePreservationTest.java

```diff
@@ -0,0 +1,236 @@
+package io.improbable.keanu.tensor;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import org.junit.Before;
+import org.junit.Test;
+import org.nd4j.linalg.api.shape.Shape;
+
+import static org.junit.Assert.assertArrayEquals;
+
+public class ScalarTensorShapePreservationTest {
+
+    private static DoubleTensor doubleTensor;
+    private static DoubleTensor scalarDoubleTensor;
+    private static IntegerTensor intTensor;
+    private static IntegerTensor scalarIntTensor;
+
+    @Before
+    public void setup() {
+        doubleTensor = DoubleTensor.create(new double[]{1, 2, 3, 4, 5, 6}, new long[]{2, 3});
+        scalarDoubleTensor = DoubleTensor.create(1.0, new long[]{1, 1, 1, 1});
+
+        intTensor = IntegerTensor.create(new int[]{5, 6, 7, 8, 9, 10}, new long[]{3, 2});
+        scalarIntTensor = IntegerTensor.create(1, new long[]{1, 1, 1});
+    }
+
+    @Test
+    public void doubleScalarMultiplicationPreservesShape() {
```

Comment:
Can we also add in tests for `scalarDoubleTensor.times(doubleTensor)` etc? And `integerTensor.times(scalarIntTensor)`?